packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Portal } from '../Portal/Portal';
+import { css, cx } from 'emotion';
+import { GrafanaTheme, ThemeContext } from '../..';
+
+const getStyles = (theme: GrafanaTheme) => ({
+  modal: css`
+    position: fixed;
+    z-index: ${theme.zIndex.modal};
+    width: 100%;
+    background: ${theme.colors.pageBg};
+    box-shadow: 0 3px 7px rgba(0, 0, 0, 0.3);
+    background-clip: padding-box;
+    outline: none;
+
+    max-width: 750px;
+    left: 0;
+    right: 0;
+    margin-left: auto;
+    margin-right: auto;
+    top: 10%;
+  `,
+  modalBackdrop: css`
+    position: fixed;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: ${theme.zIndex.modalBackdrop}
+    background-color: ${theme.colors.bodyBg}
+    opacity: 0.8;
+    backdrop-filter: blur(4px);
+  `,
+  modalHeader: css`
+    background: ${theme.background.pageHeader};
+    box-shadow: ${theme.shadow.pageHeader};
+    border-bottom: 1px soliod ${theme.colors.pageHeaderBorder};
+    display: flex;
+  `,
+  modalHeaderTitle: css`
+    font-size: ${theme.typography.heading.h3};
+    padding-top: calc(${theme.spacing.d} * 0.75);
+    margin: 0 calc(${theme.spacing.d} * 3) 0 calc(${theme.spacing.d} * 1.5);
+  `,
+  modalHeaderClose: css`
+    margin-left: auto;
+    padding: 9px ${theme.spacing.d};
+  `,
+  modalContent: css`
+    padding: calc(${theme.spacing.d} * 2);
+  `,
+});
+
+interface Props {
```

Comment:
Talking props, I consider this component as an abstraction over a modal in general. Can imagine ConfirmationModal in the future (if we don't have sth like this already) that could have UI structure built like [this ant.d's component](https://ant.design/components/modal/#header):
![image](https://user-images.githubusercontent.com/2376619/65606866-e374e000-dfab-11e9-9ea8-8fa9eb804313.png)

Maybe we could already adjust the props here to be sth like:
```
header?: string | JSX.Element;
footer?: JSX.Element;
children: // as content 
```

Not 100% sure it's the best idea as it has consequences on how we handle `onDismiss` for instance (are we delegating that to header when passed as JSX.Element, should close button be in every modal, how to render it when we have header-less modal)?

Reply:
@kay sorry for building up the scope here but since we started working on the Modal i think we could put somme UX/UI effort arround it as well and spend some time thinking about API for it. I think it can be one of the popular components in @grafana/ui 

WDYT @torkelo ?